File: examples/documentation/tests/bindings/test_documentation.rb
# frozen_string_literal: true

require 'test/unit'
require 'documentation'

include Test::Unit::Assertions

assert_equal Documentation.hello(Documentation::Pet.new("Tom")), "Hello Tom!"
assert_equal Documentation::Person.new("Daniel").get_name(), "Daniel"
assert_equal Documentation.add(2, 4), 6

File: examples/documentation/tests/bindings/test_documentation.py
import unittest
import documentation

class TestHello(unittest.TestCase):
    def test_hello(self):
        self.assertEqual(documentation.hello(documentation.Pet("Tom")),
                         "Hello Tom!", "Should be `Hello Tom!`")

class TestGetName(unittest.TestCase):
    def test_get_name(self):
        self.assertEqual(documentation.Person("Daniel").get_name(), "Daniel", "Should be Daniel")
        
class TestAdd(unittest.TestCase):
    def test_add(self):
        self.assertEqual(documentation.add(2, 4), 6, "Should be 6")


if __name__ == '__main__':
    unittest.main()
